### PR TITLE
[BUG] Fix evaluate incorrectly overriding scores of prior metrics when one metric fails

### LINKED ISSUE #5556: [BUG] Evaluate function doesn't give the right results on all scorers when using PinballLoss and RecursiveTabularRegressionForecaster
**Describe the bug**
The evaluate function doesn't always give the right results when using multiple scorers on some type of forecasters.
It seems to be mostly related to the PinballLoss scorer and the RecursiveTabularRegressionForecaster forecaster.

**To Reproduce**
Try with and without the PinballLoss as well as with one of the 2 forecasters.

```python
from sktime.datasets import load_airline
from sktime.performance_metrics.forecasting import MeanAbsolutePercentageError, MeanAbsoluteScaledError
from sktime.performance_metrics.forecasting.probabilistic import PinballLoss
from sktime.split import ExpandingWindowSplitter
from sktime.forecasting.model_evaluation import evaluate
from sklearn.linear_model import Ridge
from sktime.forecasting.compose import RecursiveTabularRegressionForecaster

df = load_airline()[:60]

scorers = [
    MeanAbsolutePercentageError(),
    MeanAbsoluteScaledError(sp=12),
    # try by commenting the pinball loss
    PinballLoss(),
]

# try by commenting one of the 2 forecaster
forecaster = RecursiveTabularRegressionForecaster(Ridge(random_state=42))
# forecaster = NaiveForecaster()

splitter = ExpandingWindowSplitter(fh=[1, 2, 3], initial_window=45, step_length=2)

evaluation = evaluate(forecaster=forecaster, cv=splitter, scoring=scorers, y=df)

print(evaluation["test_MeanAbsolutePercentageError"].mean())
print(evaluation["test_MeanAbsoluteScaledError"].mean())
```

**Additional details**
The following warning is raised by evaluate when the PinballLoss is used.
```
 In evaluate, fitting of forecaster RecursiveTabularRegressionForecaster failed,
                you can set error_score='raise' in evaluate to see
                the exception message.
                Fit failed for the 6-th data split, on training data y_train with
                cutoff NaT, and len(y_train)=57.
                The score will be set to nan.
                Failed forecaster with parameters: RecursiveTabularRegressionForecaster(estimator=Ridge(random_state=42)
```

**Expected behavior**
We should not have the same output when comparing the means of both metrics.
If it's because of a failed fit, we should have missing values.

**Versions**
<details>

System:
    python: 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
executable: /local_disk0/.ephemeral_nfs/envs/pythonEnv-e1ae0322-f241-4025-b31d-fe842e48eb5d/bin/python
   machine: Linux-5.15.0-1042-azure-x86_64-with-glibc2.35

Python dependencies:
          pip: 22.2.2
       sktime: 0.24.1
      sklearn: 1.1.1
       skbase: 0.6.1
        numpy: 1.22.4
        scipy: 1.9.1
       pandas: 1.4.4
   matplotlib: 3.5.2
       joblib: 1.2.0
        numba: 0.58.1
  statsmodels: 0.13.2
     pmdarima: 2.0.4
statsforecast: 1.6.0
      tsfresh: 0.20.1
      tslearn: 0.5.3.2
        torch: None
   tensorflow: 2.12.0
tensorflow_probability: None

</details>

### PR DESCRIPTION
**Pull request by dhairya-motta**
**Reference Issues/PRs**
Fixes #5556.

**What does this implement/fix? Explain your changes.**
The `evaluate` function in `sktime.forecasting.model_evaluation` produced identical (and incorrect) scores for all metrics when at least one metric failed during evaluation, and sometimes crashed with probabilistic metrics.

This PR fixes the exception handler in `_evaluate_window` in two major ways:
1. Previously, when an exception was caught, the fallback unconditionally assigned `temp_result[f"test_{metric.name}"] = [score]`, overwriting prior successful metrics and failing to apply `_get_pred_args_from_metric` to resolve the correct quantile keys (e.g., `_0.1`, `_0.9`). The new logic iteratively parses the requested metrics, assigning `error_score` (default `np.nan`) strictly with the proper metric names.
2. The `old_naming` logic, which dictates whether quantile tags should be appended, was previously evaluated *inside* the `try` block. If a forecaster failed during `fit()`, the code jumped to the exception block before evaluating `old_naming`. As a result, it bypassed the correct naming logic entirely, erasing quantile tags. I have fixed this by moving the `old_naming` mapping logic above the `try` block so it is safely evaluated regardless of where an exception occurs.

**Does your contribution introduce a new dependency? If yes, which one?**
No.

**What should a reviewer concentrate their feedback on?**
The updated logic in the `_evaluate_window` exception block, specifically the initialization of `old_name_mapping` before the `try` block.

**Did you add any tests for the change?**
Yes. I added `test_evaluate_probabilistic_error_score` in `test_evaluate.py` as a dedicated regression test. It utilizes a `_FailingForecaster` and multiple `PinballLoss` metrics (`alpha=0.1` and `alpha=0.9`) to explicitly verify that the output dataframe successfully resolves and populates multiple probabilistic columns when an estimator fails during `fit()`.

**PR checklist**
*For all contributions*
- [x] I've added myself to the list of contributors with any new badges I've earned :-)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

This PR contains many unrelated changes. Please make sure your PR only contain the changes described, and not accidentally changes from other PR.

**Comment by dhairya-motta** [comment]
I've cleaned up the branch and fixed the linting issues, everything should be good to go now!

File: docs/source/examples [hunk 1]
```diff
@@ -1 +1 @@
-../../examples
\ No newline at end of file
+../../examples
```
Comment (fkiraly) [inline docs/source/examples]: please revert this change too, this is unrelated

**Review by fkiraly — CHANGES_REQUESTED** [review]
* please revert unrelated changes
* please add a regression test, i.e., a test that would fail before the fix

File: docs/source/examples (hunk 1 above)
Comment (dhairya-motta) [reply docs/source/examples]: fixed the unrelated documentation/source/examples commit
most likely, it happened because my Git on Windows considers this symbolic link a text file and thats why it adds a newline to it via the end-of-file-fixer pre-commit hook, which i should have already deleted
ive even fixed this same issue on all of my PRs, sorry for the inconvenience

**Comment by dhairya-motta** [comment]
ive included the regression test using a _FailingForecaster

in the process i actually found another bug, the fallback naming was not happening if the model fails early in the fit() because it was enclosed in a try block. I have moved it outside of the try block to ensure proper resolving of column keys.

tests pass locally now, is this the right way to do this?
